src/Simulation/Common/SimulatorBase.cs

```diff
@@ -346,5 +366,16 @@ public virtual void Fail(System.Runtime.ExceptionServices.ExceptionDispatchInfo
         {
             OnFail?.Invoke(exceptionInfo);
         }
+
+        #region Stack trace collection support 
+        private StackTraceCollector stackTraceCollector = null;
+        
+        public void EnableStackTrace()
```

Comment:
I would like to see if we can instead enabling by default and just disabling it explicitly.